app/representers/cms/layout_representer.rb

```diff
@@ -11,7 +11,7 @@ module CMS::LayoutRepresenter
   property :system_name
   property :liquid_enabled
 
-  with_options(if: ->(user_options:, **) { !user_options&.dig(:short) }) do
+  with_options(if: ->(options) { !options[:user_options]&.dig(:short) }) do
```

Comment:
In original version, seems like all options but `user_options` is ignored. I see no issue with original version and think that it is fine to stay as it was. If we are to refactor, perhaps this suggestion would be more idiomatic.

```suggestion
  with_options(if: ->(**options) { !options.dig(:user_options, :short) }) do
```

Reply:
actually I see many instances where `user_options:, **` was replaced in a similar fashion. I think these should stay as they are. It makes clear we only care about that option.

Actually in Ruby 3.2 there is this use of `*`, `**` and `&` for anonymous forwarding of argument types.

See https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ArgumentsForwarding and "UseAnonymousForwarding". So these are things that are kind of used as intended IMO.

Same with all replacement of `method(*)` with `method(...)`. IMO we should leave the original version if it ain't broken.

Either way, safer to use `#dig` as intended if we keep the refactoring.

Reply:
> Same with all replacement of method(*) with method(...). IMO we should leave the original version if it ain't broken.

Regarding this, I left a comment here that explains that they are not equivalent https://github.com/3scale/porta/pull/3847/files#r1764898634

Now, I am not guaranteeing that this replacement is required in every single case - maybe in some methods were only accepting positional arguments, so this change might not be needed, but I actually think I only change the method definition where something was breaking, so I think this is actually needed.

If you prefer, I can revert that change and see if anything fails, just to make sure the change is indeed needed. But I don't see a reason why - I think using `...` is a good thing to do.

Reply:
Regarding `user_options` specifically, now that some time has passed, I don't remember why this was needed, and looking at the code, I agree with you that it seems weird and unneeded. But I am sure I wouldn't have touched it if it wasn't broken, because even when I first added that `user_options` thing, I found the documentation of the gem very confusing, and I'd have preferred not to deal with it again :grimacing: 

From the [docs](https://trailblazer.to/2.1/docs/representable/#user-options) by the way, it seems that the two ways are interchangeable, but I think the thing with Ruby 3.1 is that probably what is sent to that lambda is a single positional argument, which is a hash (we call it `options`), and not keyword arguments.

In Ruby 2 it was probably splatted automatically, but maybe in Ruby 3 it isn't?

Reply:
Consider this example:

```
lambda1 = -> (options) { puts options[:hello] }
lambda1.call({ hello: 'hi', goodbye: 'bye' })
lambda1.call(hello: 'hi', goodbye: 'bye')
```

```
lambda2 = -> (hello:, **) { puts hello }
lambda2.call({ hello: 'hi', goodbye: 'bye' })
lambda2.call(hello: 'hi', goodbye: 'bye')
```

In Ruby 2.7 all invocations for both lambdas print `hi`.

In Ruby 3.1 though all print `hi` except this one:

```
irb(main):032:0> lambda2.call({ hello: 'hi', goodbye: 'bye' })
(irb):31:in `block in <top (required)>': wrong number of arguments (given 1, expected 0; required keyword: hello) (ArgumentError)
```

And I think this is exactly the error that was observed in the original implementation.